### DISCUSSION TITLE
Repeated prefix URL

### DISCUSSION THREAD

**Discussion by mdodell** [janko/rodauth-omniauth#6] [Q&A]
Hey @janko!

When I was working on implementing Rodauth Omniauth with a [Prefix URL](https://github.com/mdodell/rodauth_omniauth_test/blob/main/app/misc/rodauth_app.rb#L8), I noticed that when my Omniauth request to `/auth/:provider`, the redirect URI included the prefix URL twice.

For example,`https://www.linkedin.com/oauth/v2/authorization?client_id=86aeer6zhrqq9e&redirect_uri=http://127.0.0.1:3000/api/v1/api/v1/auth/linkedin/callback&response_type=code&scope=r_emailaddress+r_liteprofile&state=SOME_STATE`.

As you can see, `/api/v1/api/v1` should probably just be `/api/v1`. I have a sample repository [here](https://github.com/mdodell/rodauth_omniauth_test).

**Comment by janko** [comment]
Hi, it doesn't seem that omniauth-linkedin-oauth2 strategy is currently compatible with OmniAuth 2.0. There was a [change](https://github.com/omniauth/omniauth/commit/c606a00798a7f516c37c67a708864f9a3eb9f796) with how request/callback paths are generated, which meant that `callback_path` now includes the `SCRIPT_NAME`. The LinkedIn strategy [additionally prepends](https://github.com/decioferreira/omniauth-linkedin-oauth2/blob/fe10f041ab98b5032f79883c89ad32143fa7404d/lib/omniauth/strategies/linkedin.rb#L36-L38) `SCRIPT_NAME` to callback path, leading to the duplication you're seeing.

In most common Rails installations `SCRIPT_NAME` will be empty, so this issue won't become apparent, but with the way rodauth-omniauth uses Roda routing, `SCRIPT_NAME` will be `/api/v1` in this case (the prefix will act as a "subdirectory"), so you end up with `/api/v1/api/v1`. The [fix](https://github.com/omniauth/omniauth-github/commit/f12d90fa8c06e2e63f9d242a94267e00b2618dae#diff-38ce3ba47186e72345d3d5f31f31db99b7a874c8c15ab23ac9bcd19ef38318c2L78-R78) is very simple, you could try to patch the strategy's `#callback_url` and see if that works.

```rb
      def callback_url
        full_host + callback_path
      end
```
